File: src/main/java/com/codeperfection/shipit/repository/TransporterRepository.java
package com.codeperfection.shipit.repository;

import com.codeperfection.shipit.entity.Transporter;
import com.codeperfection.shipit.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface TransporterRepository extends JpaRepository<Transporter, UUID> {

    Page<Transporter> findByUserAndIsActiveTrue(User user, Pageable pageable);

    Optional<Transporter> findByUuidAndUser(UUID uuid, User user);
}

File: src/main/kotlin/com/codeperfection/shipit/repository/TransporterRepository.kt
package com.codeperfection.shipit.repository

import com.codeperfection.shipit.entity.Transporter
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface TransporterRepository : JpaRepository<Transporter, UUID> {

    fun findByUserIdAndIsActiveTrue(userId: UUID, pageable: Pageable): Page<Transporter>

    fun findByIdAndUserIdAndIsActiveTrue(transporterId: UUID, userId: UUID): Transporter?
}
